apis/apps/v1alpha1/opsrequest_types.go

```diff
@@ -448,11 +448,26 @@ type HorizontalScaling struct {
 	// +optional
 	Instances []InstanceTemplate `json:"instances,omitempty"`
 
-	// Specifies the names of instances to be scaled down.
-	// This provides control over which specific instances are targeted for termination when reducing the replica count.
+	// Scale out or in the specified instances based on the operator.
 	//
 	// +optional
 	OfflineInstances []string `json:"offlineInstances,omitempty"`
+
+	// Replicas will be automatically synchronized based on the offline instances when it is set to true and no any replicas are specified.
+	// +optional
+	AutoSyncReplicas bool `json:"autoSyncReplicas,omitempty"`
```

Comment:
// Controls whether changes in offlineInstances automatically trigger replicas adjustments.
//
// When operator is Add and an instance is added to offlineInstances:
// - If true, replicas will be automatically decreased by the number of added instances.
// - If false, replicas will remain unchanged, and new instances will be created by the reconciliation controller loop.
//
// When operator is Delete and an instance is removed from offlineInstances:
// - If true, replicas will be automatically increased by the number of removed instances.
// - If false, replicas will remain unchanged, and the reconciliation controller loop will handle the instance deletion.

Reply:
@wangyelei seems this is a dangerous combination:

When operator is Delete and an instance is removed from offlineInstances:
If false, replicas will remain unchanged, and the reconciliation controller loop will handle the instance deletion.

Reply:
if false, controller will create the offlineInstances and remove some pods to keep replicas is unchanged.

Reply:
I think we should not provide the `autoSyncReplicas` option to avoid unexpected behavior (e.g. if user set it to false and pods're deleted automatically)
If user doesn't specify `replicas` explicitly, it should be regarded as `autoSyncReplicas` equals to true.